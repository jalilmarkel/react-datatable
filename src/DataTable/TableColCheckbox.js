import React, { memo, useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DataTableContext } from './DataTableContext';
import { CellBase } from './Cell';
import Checkbox from './Checkbox';

const TableColStyle = styled(CellBase)`
  flex: 0 0 48px;
  user-select: none;
  white-space: nowrap;
  font-size: ${props => props.theme.header.fontSize};
  color: ${props => props.theme.header.fontColor};
  height: ${props => props.theme.header.height};
`;

const TableCol = memo(({
  onClick,
}) => {
  const { selectableRowsComponent, selectableRowsComponentProps, selectedRows, allSelected } = useContext(DataTableContext);

  return (
    <TableColStyle className="rdt_TableCol">
      <Checkbox
        name="select-all-rows"
        component={selectableRowsComponent}
        componentOptions={selectableRowsComponentProps}
        onClick={onClick}
        checked={allSelected}
        indeterminate={selectedRows.length > 0 && !allSelected}
      />
    </TableColStyle>
  );
});

TableCol.propTypes = {
  onClick: PropTypes.func,
};

TableCol.defaultProps = {
  onClick: null,
};

export default TableCol;
