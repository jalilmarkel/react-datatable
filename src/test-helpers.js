import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

