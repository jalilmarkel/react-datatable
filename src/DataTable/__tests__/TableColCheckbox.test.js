import 'jest-styled-components';

import React from 'react';
import { shallowWithTheme } from '../../test-helpers';

import TableColCheckbox from '../TableColCheckbox';

test('component <TableColCheckbox /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableColCheckbox row={{}} column={{ selector: 'test' }} onClick={() => { }} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});
