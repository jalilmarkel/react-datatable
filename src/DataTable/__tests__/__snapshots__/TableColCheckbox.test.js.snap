// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  font-size: 12px;
  color: rgba(0,0,0,.54);
  height: 48px;
}

<div
  class="rdt_TableCol c0"
>
  <input
    aria-label="select-all-rows"
    name="select-all-rows"
    style="font-size: 18px; cursor: pointer; margin-left: 9px;"
    type="checkbox"
  />
</div>
`;
